consolidation/consolidate.go

```diff
@@ -0,0 +1,56 @@
+package consolidation
+
+import (
+	"gopkg.in/raintank/schema.v1"
+)
+
+func Consolidate(in []schema.Point, aggNum uint32, consolidator Consolidator) []schema.Point {
+	num := int(aggNum)
+	aggFunc := GetAggFunc(consolidator)
+	outLen := len(in) / num
+	var out []schema.Point
+	cleanLen := num * outLen // what the len of input slice would be if it was a perfect fit
+	if len(in) == cleanLen {
+		out = in[0:outLen]
+		out_i := 0
+		var next_i int
+		for in_i := 0; in_i < cleanLen; in_i = next_i {
+			next_i = in_i + num
+			out[out_i] = schema.Point{Val: aggFunc(in[in_i:next_i]), Ts: in[next_i-1].Ts}
```

Comment:
why is the output using the timestamp of the last point, and not the first?    When graphite normalizes data,  it use the TS of the first point.
https://github.com/graphite-project/graphite-web/blob/master/webapp/graphite/render/functions.py#L145-L147